### PR TITLE
MULE-10680: Add Oracle support in new DB connector

### PR DESCRIPTION
**Pull request by estebanwasinger**

### CODE REVIEW

File: extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/mysql/MySqlDbUtils.java [hunk 1]
```diff
@@ -37,25 +37,29 @@ private static String buildUrlFromAttributes(String urlPrefix, String host, Inte
       buf.append(port);
     }
     buf.append("/");
-    buf.append(database);
+    if (database != null) {
```
Comment (marianogonzalez) [inline extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/mysql/MySqlDbUtils.java]: shouldn't this be a required field?

File: extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/DbConnectionProvider.java [hunk 2]
```diff
@@ -90,7 +90,7 @@
   private DataSource dataSource;
 
   @Override
-  public final DbConnection connect() throws ConnectionException {
```
Comment (marianogonzalez) [inline extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/DbConnectionProvider.java]: why did you remove the final modifier? You didn't override this method in any CP

File: extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  private OracleConnectionParameters oracleConnectionParameters;
+
+  @Override
+  protected DbConnection createDbConnection(Connection connection) throws Exception {
+    return new OracleDbConnection(connection, super.resolveCustomTypes());
+  }
+
+  @Override
+  public Optional<DataSource> getDataSource() {
+    return Optional.empty();
```
Comment (marianogonzalez) [inline extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java]: static import

File: extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  @Override
+  public Optional<DataSourceConfig> getDataSourceConfig() {
+    return Optional.ofNullable(oracleConnectionParameters);
```
Comment (marianogonzalez) [inline extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java]: static import

File: extensions/db/src/test/java/org/mule/extension/db/integration/select/SelectMetadataOutputTestCase.java [hunk 5]
```diff
@@ -57,6 +57,7 @@ public void selectAll() throws Exception {
     assertFieldOfType(record, "ID", testDatabase.getIdFieldMetaDataType());
     assertFieldOfType(record, "POSITION", testDatabase.getPositionFieldMetaDataType());
     assertFieldOfType(record, "NAME", typeBuilder.stringType().build());
+    //The following assertion will fail in MySQL due that Blob is communicated as a byte[]
```
Comment (marianogonzalez) [inline extensions/db/src/test/java/org/mule/extension/db/integration/select/SelectMetadataOutputTestCase.java]: this is not a valid thing to do. Can you use an assumption and skip the assertion if you're using mySQL?

File: extensions/db/src/test/java/org/mule/extension/db/integration/update/UpdateTestCase.java [hunk 6]
```diff
@@ -46,6 +46,7 @@ public UpdateTestCase(String dataSourceConfigResource, AbstractTestDatabase test
     return new String[] {"integration/update/update-config.xml"};
   }
 
+  //This test will fail with MySQL due that doesn't support the MERGE statement
```
Comment (marianogonzalez) [inline extensions/db/src/test/java/org/mule/extension/db/integration/update/UpdateTestCase.java]: skip it for mysql then

**Comment by marianogonzalez** [comment]
Allure??? Where is my Allure??

File: extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/mysql/MySqlDbUtils.java (hunk 1 above)
Comment (estebanwasinger) [inline extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/mysql/MySqlDbUtils.java]: In any DB connection the specification of the database name or schema is optional, due that you can specify it later in a query

File: extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java (hunk 3 above)
Comment (estebanwasinger) [inline extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java]: In this case, the usage of the full static call enables to understand better and quickest what is going on

File: extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java (hunk 4 above)
Comment (estebanwasinger) [inline extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java]: In this case, the usage of the full static call enables to understand better and quickest what is going on

File: extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java (hunk 3 above)
Comment (marianogonzalez) [inline extensions/db/src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnectionProvider.java]: empty() is pretty clear when the return type is Optional. We usually always use a static import in this case
